src/activity_item_handling.cpp

```diff
@@ -66,6 +66,7 @@ void put_into_vehicle( player &p, const std::list<item> &items, vehicle &veh, in
             continue;
         }
         if( !veh.add_item( part, it ) ) {
+            it.charges -= veh.add_charges( part, it );
```

Comment:
Some items aren't counted by charges. I see than `add_charges()` returns zero in this case, but that's misleading and quite hackish. Also, `item::mod_charges()` is much safer than changing the value directly (not mentioning encapsulation) as it can handle various cases (not-counted-by-charges, infinite charges, negative charges, e.t.c ).